### DISCUSSION TITLE
Prisma client extension support

### DISCUSSION THREAD

**Discussion by lewebsimple** [hayes/pothos#965] [Q&A]
Hello :wave: 

Since Prisma client extensions are now generally available, I was curious as to how it would be possible to expose extended result fields.

Let's say I have the following model:
```
model Person {
  first String
  last String
}
```

... with the following result extension:
```
Prisma.defineExtension({
  name: "person-fullname",
  result: {
    person: {
      fullname: {
        needs: { first: true, last: true },
        compute: ({ first, last }) => `${first} ${last}`,
      },
    },
  },
});
```

... it's not currently possible to do this:
```
  builder.prismaNode("Person", {
    id: { field: "id" },
    fields: (t) => ({
      first: t.exposeString("first"),
      last: t.exposeString("last"),
      fullname: t.exposeString("fullname"),
    }),
  });
```

Is there a workaround for this ATM or any plan to support extended result fields ?

**Comment by hayes** [comment]
This would require a big refactor of how the Prisma plugin derives it's types.  Currently Pothos uses a generator based in your prisma schema to generate the needed type-information, and so it can't be aware of the type-information available from these client extensions.  Getting this all to work well together is likely more effort than it's worth.

You can very easily achieve something similar doing this at the GraphQL level rather than the prisma level:

```ts
builder.prismaNode("Person", {
    select: { id: true },
    id: { field: "id" },
    fields: (t) => ({
      first: t.exposeString("first"),
      last: t.exposeString("last"),
      fullname: t.string('fullname', {
        select: { first: true, last: true },
        resolve: ({ first, last }) => `${first} ${last}`,
      })
    }),
  });
```
